lms/templates/courseware/courseware.html

```
@@ -22,7 +22,7 @@
    (course.enable_proctored_exams or course.enable_timed_exams)
    )
 
-   completion_aggregator_url = settings.COMPLETION_AGGREGATOR_URL if settings.FEATURES.get("SHOW_PROGRESS_BAR", False) else ""
```

Leave this as is, since the settings are intertwined, the url should be empty if the feature is turned off.

---

`settings.COMPLETION_AGGREGATOR_URL` raises attribute error if it is not set and we are checking the feature below already so it should be enough to just check whether `COMPLETION_AGGREGATOR_URL` is set

---

That's my thinking. SHOW_PROGRESS_BAR == True doesn't work without COMPLETION_AGGEGATOR_URL.

So it's better to raise an error here if it's not defined, before the environment becomes active than discover the mistake after deploying as that will be much harder to find.

---

The attribute error is raised when a user visits lms url i.e. when the template is rendered, not at the time of deployment.

---

Cool! That makes sense. I'm happy to merge!